### PR TITLE
Handle copy failure

### PR DESCRIPTION
**Pull request by haonliu**
#### Description of changes
- When copy failed, show toast with text: Failed to copy Failure details. Please try again.
- Removed Contextual Menu and used menu props on action button component as a part of this PR, so that we do not have to handle the state of the contextual menu.
- Updated snapshots and unit tests.

image

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/common/components/cards/card-kebab-menu-button.tsx [hunk 1]
```diff
@@ -26,10 +25,9 @@ export type CardKebabMenuButtonDeps = {
 } & IssueFilingButtonDeps;
 
 export interface CardKebabMenuButtonState {
-    isContextMenuVisible: boolean;
     showNeedsSettingsContent: boolean;
-    target?: HTMLElement | string | MouseEvent | IPoint | null;
     showingCopyToast: boolean;
+    toastText: string;
```
Comment (AhmedAbdoOrtiga) [inline src/common/components/cards/card-kebab-menu-button.tsx]: I think it'd better if we extract the toast component in a separate and toasText should be a prop, that would help even refactoring with the copy-issue-details **button**

File: src/common/components/cards/card-kebab-menu-button.tsx [hunk 2]
```diff
@@ -163,11 +161,15 @@ export class CardKebabMenuButton extends React.Component<CardKebabMenuButtonProp
...
+    }
+
+    private hideToast(): void {
+        this.setState({ showingCopyToast: false, toastText: '' });
+    }
 
-    private dismissDropdown(): void {
-        this.setState({ target: null, isContextMenuVisible: false });
+    private showToastWithFailureMessage(): void {
+        this.setState({ showingCopyToast: true, toastText: 'Failed to copy failure details. Please try again.' });
```
Comment (AhmedAbdoOrtiga) [inline src/common/components/cards/card-kebab-menu-button.tsx]: Can we check with Fer and Chelsea since this is a UI change

Comment (lisli1) [reply src/common/components/cards/card-kebab-menu-button.tsx]: I am also curious if we have asked Fer and Chelsea what they think of this failure message.

**Comment by lisli1** [comment]
Does the UI look the same without Contextual Menu?

File: src/common/components/cards/card-kebab-menu-button.tsx (hunk 2 above)
Comment (haonliu) [reply src/common/components/cards/card-kebab-menu-button.tsx]: yeah, I've checked with Fer yesterday. This is the text she told me yesterday.

File: src/common/components/cards/card-kebab-menu-button.tsx (hunk 1 above)
Comment (haonliu) [reply src/common/components/cards/card-kebab-menu-button.tsx]: Agreed, but that itself could be a separate PR.

**Comment by haonliu** [comment]
> Does the UI look the same without Contextual Menu?

Yes. This seems to be the new way to do a contextual menu in office fabric: https://developer.microsoft.com/en-us/fabric#/controls/web/contextualmenu

File: src/common/components/cards/card-kebab-menu-button.tsx [hunk 3]
```diff
@@ -170,11 +162,15 @@ export class CardKebabMenuButton extends React.Component<CardKebabMenuButtonProp
...
 
-    private openDropdown = (event): void => {
-        this.setState({ target: event.currentTarget, isContextMenuVisible: true });
-    };
+    private showToastWithSuccessMessage(): void {
+        this.setState({ showingCopyToast: true, toastText: 'Failure details copied.' });
+    }
+
+    private hideToast(): void {
+        this.setState({ showingCopyToast: false, toastText: '' });
```
Comment (dbjorge) [inline src/common/components/cards/card-kebab-menu-button.tsx]: Does the toast's disappearance involve a fade out animation, or is it immediate? If it isn't immediate, we might not want to empty the text here

Comment (haonliu) [reply src/common/components/cards/card-kebab-menu-button.tsx]: I tried remove the text state change here, there is no change to the dismiss animation. So my understanding is that there is no fade out animation. 
Also, compared with master, not seeing fade out animation.

**Pull request merged**